exercises/forall-param-reductions_solution.cpp

```diff
@@ -0,0 +1,293 @@
+//~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~//
```

Comment:
@mdavis36 are you planning to add a tutorial section in the User Guide for this? If not, would it be better to move this file into the RAJA/examples directory?

Reply:
Sure I'll move it to examples for now, I am adding some documentation to the reductions page however.

Reply:
Done.